extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/handler/ZonePlayerHandler.java

```diff
@@ -769,20 +770,22 @@ protected void updateTrackMetaData() {
             String artist = null;
             String title = null;
             String album = null;
+            
             if (currentTrack != null) {
-                if (currentTrack.getAlbumArtist().equals("")) {
-                    artist = currentTrack.getCreator();
-                } else {
-                    artist = currentTrack.getAlbumArtist();
-                }
+                artist = !currentTrack.getAlbumArtist().isEmpty() ? currentTrack.getAlbumArtist()
+                        : currentTrack.getCreator();
 
                 if (!currentTrack.getTitle().contains("x-sonosapi-stream")) {
                     title = currentTrack.getTitle();
+                } else if (opmlUrl == null && currentUriMetaData != null) {
```

Comment:
This breaks the final else I just added that must be only if title contains x-sonosapi-stream.

I would propose:
```
                 if (!currentTrack.getTitle().contains("x-sonosapi-stream")) {
                     title = currentTrack.getTitle();
                 } else {
                     if (opmlUrl == null && currentUriMetaData != null) {
                         artist = currentUriMetaData.getTitle();
                         title = currentTrack.getStreamContent();
                     } else {
                         // For tune-in, don't reset the title as it was previously set in updateCurrentURIFormatted
                         // using the Opml HTTP query
                         title = stateMap.get("CurrentTitle");
                     }
                 }
```